backend/src/nft/nft.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from "@nestjs/common";    
+import { ProviderService } from 'src/shared/services/provider/provider.service';
+import { SignerService } from 'src/shared/services/signer/signer.service';
+import { ethers } from 'ethers';
+import * as NFTContract from '../assets/contracts/ApeToken.sol/ApeToken.json';
+
+@Injectable()
+export class NFTService {
+    nftContract: ethers.Contract;
+    
+    async metadata(tokenId: number) {
```

Comment:
you don't have any nftContract instance

Reply:
Ok, I have just added the contract instance . Thanks @allen-hsu